datahub-web-react/src/app/auth/SignUp.tsx

```diff
@@ -0,0 +1,204 @@
+                        >
+                            <FormInput prefix={<UserOutlined />} data-testid="name" />
+                        </Form.Item>
+                        <Form.Item
+                            rules={[
+                                { required: true, message: 'Please fill in your password!' },
+                                ({ getFieldValue }) => ({
+                                    validator() {
+                                        if (getFieldValue('password').length < 8) {
+                                            return Promise.reject(
+                                                new Error('Your password is less than 8 characters!'),
+                                            );
+                                        }
+                                        return Promise.resolve();
+                                    },
+                                }),
+                            ]}
+                            name="password"
+                            // eslint-disable-next-line jsx-a11y/label-has-associated-control
+                            label={<label style={{ color: 'white' }}>Password</label>}
+                        >
+                            <FormInput prefix={<LockOutlined />} type="password" data-testid="password" />
+                        </Form.Item>
+                        <Form.Item
+                            rules={[
+                                { required: true, message: 'Please confirm your password!' },
+                                ({ getFieldValue }) => ({
+                                    validator() {
+                                        if (getFieldValue('confirmPassword') !== getFieldValue('password')) {
+                                            return Promise.reject(new Error('Your passwords do not match!'));
+                                        }
+                                        return Promise.resolve();
+                                    },
+                                }),
+                            ]}
+                            name="confirmPassword"
+                            // eslint-disable-next-line jsx-a11y/label-has-associated-control
+                            label={<label style={{ color: 'white' }}>Confirm Password</label>}
+                        >
+                            <FormInput prefix={<LockOutlined />} type="password" data-testid="confirmPassword" />
+                        </Form.Item>
+                        <Form.Item
+                            rules={[{ required: true, message: 'Please fill in your role!' }]}
+                            name="role"
+                            // eslint-disable-next-line jsx-a11y/label-has-associated-control
+                            label={<label style={{ color: 'white' }}>Select your role!</label>}
+                        >
+                            <RoleSelector placeholder="Role">
```

Comment:
I know we call this "Role" but really its more like Title

Reply:
Refactored :)